### PR TITLE
BridgePool & BridgePoolFactory

### PR DESCRIPTION
**Pull request by gusjavaz**
## Scope
https://madhive.atlassian.net/browse/MP-269
https://madhive.atlassian.net/browse/MP-268
This PR includes merge of https://github.com/MadBase/MadNet/pull/126 with some changes

### CODE REVIEW

File: bridge/contracts/BridgePool.sol [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+            )
+        );
+        require(
+            merkleProof.checkProof(bClaims.stateRoot, merkleProof.computeLeafHash()),
+            string(abi.encodePacked(BridgePoolErrorCodes.BRIDGEPOOL_COULD_NOT_VERIFY_PROOF_OF_BURN))
+        );
+        _safeTransferERC20(IERC20Transferable(_erc20TokenContract), msg.sender, burnedUTXO.value);
+    }
+
+    receive() external payable {}
```
Comment (z-j-lin) [inline bridge/contracts/BridgePool.sol:138]: Locked eth. This contract needs a method to withdraw eth if it can receive eth.

**Review by z-j-lin — CHANGES_REQUESTED** [review]

File: bridge/contracts/BridgePool.sol (hunk 1 above)
Comment (gusjavaz) [reply bridge/contracts/BridgePool.sol:138]: Fixed in latest commit. Now the ETH obtained from BToken burning is immediately transferred to Foundation contract. Thanks @z-j-lin

File: .gitignore [hunk 2]
```diff
@@ -25,4 +26,14 @@ local-geth/*
 
 # generated files
 scripts/generated
-node_modules/
\ No newline at end of file
+localstate.swagger.json
```
Comment (Cr0wn-Gh0ul) [inline .gitignore]: Are we sure these need to be ignored?

Comment (gusjavaz) [reply .gitignore]: Sorry, this is coming from merging with Kasper's PR branch (events). @kasvtv Are you ok to include this change in gitignore (and some others related to *gen.go) in this PR or you prefer to have them in your own?

File: bridge/contracts/BridgePool.sol (hunk 1 above)
Comment (gusjavaz) [reply bridge/contracts/BridgePool.sol:138]: According to the latest guidelines, the remaining ETH from the burning of fees is transferred to BToken through a selfdestructing contract.

File: .gitignore (hunk 2 above)
Comment (gusjavaz) [reply .gitignore]: Removed legacy gitignore lines on last commit

**Pull request closed without merging**